src/shared/services/app-globals.service.ts

```diff
@@ -22,13 +22,15 @@
 
 import { Injectable } from '@angular/core';
 import { ReplaySubject } from 'rxjs/ReplaySubject';
-
+import { Map } from 'immutable';
 import { SchemaValidationErrors } from '../interfaces';
 
 @Injectable()
 export class AppGlobalsService {
   private _globalErrorsSubject: ReplaySubject<SchemaValidationErrors> = new ReplaySubject<any>(1);
   public adminMode = false;
+  public activeTabName = 'Main';
+  public firstTopLevelElementsPathMap = Map<string, string>();
```

Comment:
Doesn't have to be `Immutable.Map` or even `Map`, I think  just `{ [string: tabName]: string }` is good enough.